homeassistant/components/velbus/light.py

```diff
@@ -0,0 +1,82 @@
+"""Support for Velbus light dimmers."""
+import logging
+
+from velbus.util import VelbusException
+
+from homeassistant.components.light import (
+    ATTR_BRIGHTNESS,
+    ATTR_TRANSITION,
+    SUPPORT_BRIGHTNESS,
+    SUPPORT_TRANSITION,
+    Light,
+)
+
+from . import VelbusEntity
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+    """Old way."""
+    pass
+
+
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Set up Velbus light based on config_entry."""
+    cntrl = hass.data[DOMAIN][entry.entry_id]["cntrl"]
+    modules_data = hass.data[DOMAIN][entry.entry_id]["light"]
+    entities = []
+    for address, channel in modules_data:
+        module = cntrl.get_module(address)
+        entities.append(VelbusLight(module, channel))
+    async_add_entities(entities)
+
+
+class VelbusLight(VelbusEntity, Light):
+    """Representation of a light dimmer."""
+
+    @property
+    def supported_features(self):
+        """Flag supported features."""
+        return SUPPORT_BRIGHTNESS | SUPPORT_TRANSITION
+
+    @property
+    def is_on(self):
+        """Return true if the dimmer is on."""
+        return self._module.is_on(self._channel)
+
+    @property
+    def brightness(self):
+        """Return the brightness of the light."""
+        return self._module.get_dimmer_state(self._channel)
+
+    def turn_on(self, **kwargs):
+        """Instruct the Velbus light to turn on."""
+        if ATTR_TRANSITION in kwargs:
+            transitiontime = int(kwargs[ATTR_TRANSITION])
+        else:
+            transitiontime = 0
+        try:
+            if ATTR_BRIGHTNESS in kwargs:
+                self._module.set_dimmer_state(
+                    self._channel, int(kwargs[ATTR_BRIGHTNESS]), transitiontime,
```

Comment:
Brightness is already validated as an integer by the service schema. We don't need to copy to integer.

Reply:
The same applies to `ATTR_TRANSITION`
Both fixed in [this](https://github.com/home-assistant/home-assistant/pull/30184/commits/41a319f2011d4ff77c69b2af3b8f3e2b2c65e2b6) commit